core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandler.java

```diff
@@ -431,27 +433,29 @@ public void onThrottleFailure(@NonNull RequestThrottlingException error) {
   private void setFinalError(Statement<?> statement, Throwable error, Node node, int execution) {
     DriverExecutionProfile executionProfile =
         Conversions.resolveExecutionProfile(statement, context);
+    ExecutionInfo executionInfo = null;
     if (error instanceof DriverException) {
-      ((DriverException) error)
-          .setExecutionInfo(
-              new DefaultExecutionInfo(
-                  statement,
-                  node,
-                  startedSpeculativeExecutionsCount.get(),
-                  execution,
-                  errors,
-                  null,
-                  null,
-                  true,
-                  session,
-                  context,
-                  executionProfile));
+      executionInfo =
```

Comment:
is there a reason we only build and pass along the `executionInfo` for `DriverException` errors?

Reply:
For reasons unknown to me, the existing code was only creating a new `ExecutionInfo` for `DriverErrors`.  I suppose this is because `DriverException` has a setter for it.  Now I suppose we have another use for the `ExecutionInfo` so I can change it to create one for all `Throwables`.